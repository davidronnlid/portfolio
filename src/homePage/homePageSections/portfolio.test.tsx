import renderer from 'react-test-renderer'
import Portfolio from './portfolio'

// Basic Jest-only snapshot test
it('renders the portfolio page section correctly', () => {
    const tree = renderer.create(<Portfolio />).toJSON()
    expect(tree).toMatchSnapshot()
})

// Basic React Testing Library tests for the portfolio page section
import { getByAltText, render, screen, within } from '@testing-library/react'

describe('two projects are rendered in the portfolio, each with a related icon with a GitHub project link and an image that has a relevant project link', () => {
    beforeEach(() => {
        render(<Portfolio />)
    })

    test('Project and container device images are rendered', () => {
        // Note that the casing of the first letters of the image alt text project titles are crucial for matching only the project images and not the container images below (the container images have alt texts including the same exact strings but with the project title set to lower case)

        const desktopContainerImg = screen.getByAltText('desktop', {
            exact: false,
        })

        const desktopProjectImg = screen.getByAltText(
            'Davids Movie Browser project.',
            {
                exact: true,
            }
        )

        const mobileContainerImg = screen.getByAltText('mobile', {
            exact: false,
        })

        const mobileProjectImg = screen.getByAltText(
            'Davids Calculator project.',
            {
                exact: true,
            }
        )

        const projectAndContainerImgs = [
            desktopContainerImg,
            desktopProjectImg,
            mobileContainerImg,
            mobileProjectImg,
        ]

        expect(projectAndContainerImgs.every((elm) => typeof elm === 'string'))
        expect(projectAndContainerImgs.length).toEqual(4)
        projectAndContainerImgs.forEach((elm) => expect(elm).toBeVisible())
    })

    test('project image links are rendered', () => {
        const desktopProjectImg = screen.getByAltText(
            'Davids Movie Browser project.',
            {
                exact: true,
            }
        )

        const mobileProjectImg = screen.getByAltText(
            'Davids Calculator project.',
            {
                exact: true,
            }
        )

        const desktopProjectImgLink = desktopProjectImg.closest('a')

        const mobileProjectImgLink = mobileProjectImg.closest('a')

        const projectImgLinks = [desktopProjectImgLink, mobileProjectImgLink]

        projectImgLinks.forEach((elm) =>
            expect(elm).toHaveAttribute(
                'href',
                expect.stringContaining('https://davidronnlid')
            )
        )
    })

    test('a relevant Project Tooltip Github Link is rendered for each project', () => {
        const projectGithubTooltipLinks = screen.getAllByLabelText(
            'Project Tooltip Github Link'
        )

        expect(projectGithubTooltipLinks.length).toEqual(2)

        const githubLinks = screen.getAllByTestId('github-link')

        expect(githubLinks.length).toEqual(2)

        githubLinks.forEach((link) =>
            expect(link).toHaveAttribute(
                'href',
                expect.stringContaining('https://github.com/davidronnlid')
            )
        )
    })
})
