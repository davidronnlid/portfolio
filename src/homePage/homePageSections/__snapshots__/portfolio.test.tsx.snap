// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the portfolio page section correctly 1`] = `
<div
  className="MuiBox-root css-0"
  id="portfolio"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-5uhzgc-MuiTypography-root"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-8 css-am41oj-MuiGrid-root"
    >
      <button
        aria-label="Project Tooltip Github Link"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          data-testid="github-link"
          href="https://github.com/davidronnlid/movie-project"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <svg
            className="projectGithubIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--secondary-color)",
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
            />
          </svg>
        </a>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        data-cy="movie-browser-project-link"
        href="https://davidronnlidmovies.netlify.app"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="projectContainer MuiBox-root css-0"
        >
          <img
            alt="Davids Movie Browser project."
            className="desktopProjectImg"
            src="dRMScreenshot.png"
          />
          <img
            alt="Desktop screen displays Davids movie browser project."
            className="desktopContainerImg"
            src="Mac.png"
          />
        </div>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
    >
      <button
        aria-label="Project Tooltip Github Link"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          data-testid="github-link"
          href="https://github.com/davidronnlid/calculator-app"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <svg
            className="projectGithubIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--secondary-color)",
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
            />
          </svg>
        </a>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        data-cy="calculator-project-link"
        href="https://davidronnlidcalculator.netlify.app"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="projectContainer mobileProjectContainer MuiBox-root css-0"
        >
          <img
            alt="Davids Calculator project."
            className="mobileProjectImg"
            loading="lazy"
            src="calcAppScreenshot.png"
          />
          <img
            alt="Mobile screen displays Davids calculator project."
            className="mobileContainerImg"
            loading="lazy"
            src="iPhone.png"
          />
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`two projects are rendered in the portfolio, each with a related icon with a GitHub project link and an image that has a relevant project link renders the portfolio page section correctly 1`] = `
<div
  className="MuiBox-root css-0"
  id="portfolio"
>
  <h3
    className="MuiTypography-root MuiTypography-h3 css-5uhzgc-MuiTypography-root"
  />
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-8 css-am41oj-MuiGrid-root"
    >
      <button
        aria-label="Project Github Link"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          data-testid="github-link"
          href="https://github.com/davidronnlid/movie-project"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <svg
            className="projectGithubIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--secondary-color)",
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
            />
          </svg>
        </a>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        data-cy="movie-browser-project-link"
        href="https://davidronnlidmovies.netlify.app"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="projectContainer MuiBox-root css-0"
        >
          <img
            alt="Davids Movie Browser project."
            className="desktopProjectImg"
            src="dRMScreenshot.png"
          />
          <img
            alt="Desktop screen displays Davids movie browser project."
            className="desktopContainerImg"
            src="Mac.png"
          />
        </div>
      </a>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
    >
      <button
        aria-label="Project Github Link"
        aria-labelledby={null}
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        data-mui-internal-clone-element={true}
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          data-testid="github-link"
          href="https://github.com/davidronnlid/calculator-app"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <svg
            className="projectGithubIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "var(--secondary-color)",
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.012 8.012 0 0 0 16 8c0-4.42-3.58-8-8-8z"
            />
          </svg>
        </a>
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        data-cy="calculator-project-link"
        href="https://davidronnlidcalculator.netlify.app"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="projectContainer mobileProjectContainer MuiBox-root css-0"
        >
          <img
            alt="Davids Calculator project."
            className="mobileProjectImg"
            loading="lazy"
            src="calcAppScreenshot.png"
          />
          <img
            alt="Mobile screen displays Davids calculator project."
            className="mobileContainerImg"
            loading="lazy"
            src="iPhone.png"
          />
        </div>
      </a>
    </div>
  </div>
</div>
`;
